src/openrct2/ride/Vehicle.cpp

```diff
@@ -7273,33 +7276,38 @@ void Vehicle::UpdateAnimationAnimalFlying()
     }
 }
 
+uint8_t Vehicle::GetTargetFrame(rct_ride_entry_vehicle* vehicleEntry)
+{
+    // Similar to a modulus, but using a mask and multiplying before a final bit shift to get rid of the mask bits. You are left
+    // with a value from 0 to AnimationFrames - 1. Prevents skipped frames on non powers-of-two AnimationFrames.
+    return (((animationState >> std::clamp(vehicleEntry->AnimationSpeed - UINT8_WIDTH, 0, UINT32_WIDTH))
+             & std::numeric_limits<uint8_t>::max())
```

Comment:
Using numeric limits actually makes this less legible in my opinion, but who am I to read my own code?

Reply:
```suggestion
                 return (((animationState >> std::clamp(
                  vehicleEntry->AnimationSpeed - std::numeric_limits<uint8_t>::digits, 0,
                  std::numeric_limits<uint32_t>::digits))
             & std::numeric_limits<uint8_t>::max())
            * vehicleEntry->AnimationFrames)
        >> std::numeric_limits<uint8_t>::digits;
```
Here it is in full C++ incomprehensibility

Reply:
> Using numeric limits actually makes this less legible in my opinion, but who am I to read my own code?

No one forced you do use it in a horrible way. Just split up the single line into multiple statements and it will become a lot easier already, for the limits you can define constexpr constants.